File: sdk/ios/Sources/KsealSDK/Probes/NetworkRiskDetector.swift
import Foundation

/// Detects network-interception posture: a configured system HTTP proxy (the
/// most common MITM enabler reachable via public APIs on iOS). Pinning failures
/// are reported separately by the transport layer via
/// `KsealSDK.reportPinningFailure()`.
struct NetworkRiskDetector: Probe {
    let id = "network"
    private let env: DeviceEnvironment

    init(_ env: DeviceEnvironment) { self.env = env }

    func evaluate() -> Set<RiskSignal> {
        if let host = env.proxyHost(), !host.isEmpty {
            return [.proxy, .networkMitm]
        }
        return []
    }
}

File: sdk/android/src/main/kotlin/io/kseal/sdk/probes/NetworkRiskDetector.kt
package io.kseal.sdk.probes

import io.kseal.sdk.RiskSignal

/**
 * Detects network-interception posture: a configured system HTTP proxy and
 * user-installed CA certificates (the two most common MITM enablers). Pinning
 * failures are reported separately by the transport layer via
 * [io.kseal.sdk.KsealSDK.reportPinningFailure].
 */
internal class NetworkRiskDetector(private val env: DeviceEnvironment) : Probe {

    override val id: String = "network"

    override fun evaluate(): Set<RiskSignal> {
        val signals = LinkedHashSet<RiskSignal>()

        if (!env.httpProxyHost().isNullOrBlank()) {
            signals += RiskSignal.PROXY
            signals += RiskSignal.NETWORK_MITM
        }
        if (env.userInstalledCaCount() > 0) {
            signals += RiskSignal.USER_CA
            signals += RiskSignal.NETWORK_MITM
        }
        return signals
    }
}
